libs/langchain/langchain/document_loaders/parsers/pdf_docai.py

```diff
@@ -0,0 +1,217 @@
+        gcs_output_path: Optional[str] = None,
+    ):
+        from google.api_core.client_options import ClientOptions
+
+        self._options = ClientOptions(
+            api_endpoint=f"{location}-documentai.googleapis.com"
+        )
+        self._processor_name = processor_name
+        self._name = self.client.common_location_path(project=project, location="us")
+        self._gcs_output_path = gcs_output_path
+
+    def lazy_parse(self, blob: Blob) -> Iterator[Document]:
+        """Parses a blob lazily.
+
+        This is a long-running operations! A recommended way is to batch
+            documents together and use `batch_parse` method.
+        """
+        yield from self.batch_parse(
+            [blob], self._gcs_output_path  # type: ignore[arg-type]
+        )
+
+    def batch_parse(
+        self, blobs: List[Blob], gcs_output_path: str
+    ) -> Iterator[Document]:
+        """Parses a list of blobs lazily.
+
+        Args:
+            blobs: a list of blobs to parse
+            gcs_output_path: a path on GCS to store parsing results
+        This is a long-running operations! A recommended way is to decouple
+            parsing from creating Langchain Documents:
+            >>> operations = parser.docai_parse(blobs, gcs_path)
+            >>> parser.is_running(operations)
+            You can get operations names and save them:
+            >>> names = [op.name for op in operations]
+            And when all operations are finished, you can use their results:
+            >>> operations = parser.operations_from_names(operation_names)
+            >>> results = parser.get_results(operations)
+            >>> docs = parser.batch_parse_from_results(results)
+        """
+        operations = self.docai_parse(blobs, gcs_output_path=gcs_output_path)
+        is_running = self.is_running(operations)
+        while is_running:
+            is_running = self.is_running(operations)
+            if not is_running:
+                break
+            time.sleep(60)
+            print(".", end="")
+
+        results = self.get_results(operations=operations)
+        yield from self.batch_parse_from_results(results)
+
+    def batch_parse_from_results(
+        self, results: List[DocAIParsingResults]
+    ) -> Iterator[Document]:
+        for result in results:
+            for doc in self.parse_from_result(result):
+                yield doc
+
+    def parse_from_result(self, result: DocAIParsingResults) -> List[Document]:
+        """Creates a list of Documents from Google DocAI parsing results."""
+        from google.cloud.documentai_toolbox.wrappers.document import _get_shards
+        from google.cloud.documentai_toolbox.wrappers.page import _text_from_layout
+
+        output_gcs = result.parsed_path.split("/")
+        gcs_bucket_name = output_gcs[2]
+        gcs_prefix = "/".join(output_gcs[3:]) + "/"
+        shards = _get_shards(gcs_bucket_name, gcs_prefix)
+        docs, i = [], 0
```

Comment:
nit: variable name for `i` -> `page_number`

Reply:
Done